### PR TITLE
Travis CI - Python 2 build

### PR DESCRIPTION
**Pull request by cipherboy**
This PR does three things:

 - Creates a python2 docker image based on Fedora
 - Adds the python2 docker image to Travis
 - Creates a new Dockerfiles directory, moving all of the `Dockerfile*` files into it.

This will help to ensure we don't break Python2 support in the future.

### CODE REVIEW

**Comment by mpreisler** [comment]
> One suggestion would be that we could reduce the python2 to only run the unit tests, and not all of ctest. Thoughts?

I'd run it all, there are subtle differences between py2 and py3 even in the content built. Ordering, encoding, etc...

File: .travis.yml [hunk 1]
```diff
@@ -9,11 +9,16 @@ matrix:
...
+            - docker build --no-cache --tag ssg:latest -f Dockerfiles/fedora_python3 .
           script:
             - docker run ssg:latest
         - env: BASE_IMAGE="Ubuntu"
           before_install:
-            - docker build --no-cache --tag ssg_ubuntu:latest -f Dockerfile_ubuntu .
+            - docker build --no-cache --tag ssg_ubuntu:latest -f Dockerfiles/ubuntu .
           script:
             - docker run ssg_ubuntu:latest
+        - env: BASE_IMAGE="Fedora / Python 2"
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply .travis.yml]: @redhatrises Okie dokie, updated.

**Comment by cipherboy** [comment]
@redhatrises I updated it to use `BASE_IMAGE` as the path to the dockerfile and such. What do you think? :)

**Comment by redhatrises** [comment]
@cipherboy looks good.

**Comment by cipherboy** [comment]
@redhatrises I suppose we could, but that'd require maintaining (what are essentially) stock images + minimal dependencies. We will still need to pull down all of the PR's source code, build it, run the tests, which is the bulk (~90%) of our time. I think we should think about improving our build and test process, rather than expending energy to maintain clones of upstream images.

**Comment by mpreisler** [comment]
ACK

**Pull request merged**